cloud_utils/scheduler/kubernetes_connector.py

```diff
@@ -146,6 +210,36 @@ def _pod_manifest(
     )
 
 
+def _make_job_spec(
```

Comment:
this function's signature is unnecessarily complex. it can simply get the output of `_pod_manifest` and this will allow to remove 95% of the arguments

same goes for `create_cron_job`